### PR TITLE
Convert BitmapUtils.java to BitmapUtils.kt

### PR DESCRIPTION
**Pull request by amilcar-andrade**

### CODE REVIEW

**Comment by amilcar-andrade** [comment]
@jrodbx pls review if you have some time, thanks champ!

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+import androidx.annotation.DrawableRes
+import androidx.annotation.RequiresApi
+import okio.*
+import java.io.IOException
+import java.nio.ByteBuffer
+import kotlin.math.floor
+
+object BitmapUtils {
+
+    internal class ExceptionCatchingSource(delegate: Source?) : ForwardingSource(delegate!!) {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:36]: nit: move to bottom of class declaration

Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:36]:
```suggestion
    internal class ExceptionCatchingSource(delegate: Source) : ForwardingSource(delegate) {
```

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+        var thrownException: IOException? = null
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:37]:
```suggestion
        var thrownException: IOException? = null

```

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            options.inInputShareable = data.purgeable
+            options.inPurgeable = data.purgeable
+            if (data.config != null) {
+                options.inPreferredConfig = data.config
+            }
+        }
+        return options
+    }
+
+    fun requiresInSampleSize(options: BitmapFactory.Options?): Boolean {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:78]:
```suggestion
    @JvmStatic
    fun requiresInSampleSize(options: BitmapFactory.Options?): Boolean {
```

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    /**
+     * Decode a byte stream into a Bitmap. This method will take into account additional information
+     * about the supplied request in order to do the decoding efficiently (such as through leveraging
+     * `inSampleSize`).
+     */
+    @Throws(IOException::class)
+    @JvmStatic
+    fun decodeStream(source: Source?, request: Request): Bitmap {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:124]:
```suggestion
    fun decodeStream(source: Source, request: Request): Bitmap {
```

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+    private fun decodeStreamPreP(request: Request, bufferedSource: BufferedSource): Bitmap {
+        val isWebPFile = Utils.isWebPFile(bufferedSource)
+        val isPurgeable = request.purgeable && VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP
+        val options = createBitmapOptions(request)
+        val calculateSize = requiresInSampleSize(options)
+        val bitmap: Bitmap?
+        // We decode from a byte array because, a) when decoding a WebP network stream, BitmapFactory
+        // throws a JNI Exception, so we workaround by decoding a byte array, or b) user requested
+        // purgeable, which only affects bitmaps decoded from byte arrays.
+        bitmap = if (isWebPFile || isPurgeable) {
```
Comment (jrodbx) [inline picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:150]: merge declaration with assignment.  comment above

**Review by jrodbx — APPROVED** [review]

File: picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt (hunk 2 above)
Comment (amilcar-andrade) [reply picasso/src/main/java/com/squareup/picasso3/BitmapUtils.kt:37]: Whats the suggestion here? Seems the same.

**Comment by amilcar-andrade** [comment]
@jrodbx addressed your comments, pls re-review.

**Pull request merged**